sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -294,7 +296,12 @@ object ShuffleExchangeExec {
           sorter.sort(iter.asInstanceOf[Iterator[UnsafeRow]])
         }
       } else {
-        rdd
+        part match {
+          case partitioner: RangePartitioner[InternalRow @unchecked, _]
+            if partitioner.getSampledArray != null =>
+            sparkContext.parallelize(partitioner.getSampledArray.toSeq)
```

Comment:
This should parallelize the sampled array to required partition number too.

Reply:
change to : sparkContext.parallelize(partitioner.getSampledArray.toSeq, rdd.getNumPartitions)